java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightJdbcFlightStreamResultSet.java

```diff
@@ -162,45 +164,49 @@ private void executeForCurrentFlightStream() throws SQLException {
 
   @Override
   public boolean next() throws SQLException {
-    if (currentVectorSchemaRoot == null) {
-      return false;
-    }
-    while (true) {
-      final boolean hasNext = super.next();
-      final int maxRows = statement != null ? statement.getMaxRows() : 0;
-      if (maxRows != 0 && this.getRow() > maxRows) {
-        if (statement.isCloseOnCompletion()) {
-          statement.close();
-        }
+    try {
+      if (currentVectorSchemaRoot == null) {
         return false;
       }
+      while (true) {
+        final boolean hasNext = super.next();
+        final int maxRows = statement != null ? statement.getMaxRows() : 0;
+        if (maxRows != 0 && this.getRow() > maxRows) {
+          if (statement.isCloseOnCompletion()) {
+            statement.close();
+          }
+          return false;
+        }
 
-      if (hasNext) {
-        return true;
-      }
+        if (hasNext) {
+          return true;
+        }
 
-      if (currentFlightStream != null) {
-        currentFlightStream.getRoot().clear();
-        if (currentFlightStream.next()) {
-          executeForCurrentFlightStream();
-          continue;
+        if (currentFlightStream != null) {
+          currentFlightStream.getRoot().clear();
+          if (currentFlightStream.next()) {
+            executeForCurrentFlightStream();
+            continue;
+          }
+
+          flightStreamQueue.enqueue(currentFlightStream);
         }
 
-        flightStreamQueue.enqueue(currentFlightStream);
-      }
+        currentFlightStream = getNextFlightStream(false);
 
-      currentFlightStream = getNextFlightStream(false);
+        if (currentFlightStream != null) {
+          executeForCurrentFlightStream();
+          continue;
+        }
 
-      if (currentFlightStream != null) {
-        executeForCurrentFlightStream();
-        continue;
-      }
+        if (statement != null && statement.isCloseOnCompletion()) {
+          statement.close();
+        }
 
-      if (statement != null && statement.isCloseOnCompletion()) {
-        statement.close();
+        return false;
       }
-
-      return false;
+    } catch (final FlightRuntimeException e) {
+      throw FlightToJDBCExceptionMapper.map(e, "Failed to access the resource: ");
```

Comment:
Can we remove the trailing `: ` on all exceptions please?

Reply:
Done.